src/ExceptionTransformer.ts

```diff
@@ -46,9 +48,80 @@ class ExceptionTransformer {
         exception.fallback_message
       );
     }
-
     return exceptionMap;
   }
+
+  public generateSpesificFieldError(errorInfo: Exception | null | undefined) {
+    const detail =
+      errorInfo && typeof errorInfo.detail === "object" && errorInfo.detail;
+
+    if (detail) {
+      return function getError(fieldName: string) {
+        return detail[fieldName];
+      };
+    } else {
+      return () => {};
+    }
+  }
+
+  public generateErrorMessage(
+    errorInfo: Exception,
+    options: Options = {}
+  ): string {
+    let finalMessage = "";
+    const { knownErrorKeys = [], skipTypes = [] } = options;
+
+    try {
+      if (!(skipTypes && skipTypes.includes(errorInfo.type))) {
+        const errorDetail = errorInfo.detail;
+        let shouldDisplayFallbackMessage = false;
+        let message = "";
+
+        if (errorDetail && typeof errorDetail === "object") {
+          const errorDetailKeys = Object.keys(errorDetail);
+
+          if (
+            errorDetailKeys.includes("non_field_errors") &&
+            errorDetail.non_field_errors
+          ) {
+            if (typeof errorDetail.non_field_errors[0] === "string") {
+              message = errorDetail.non_field_errors[0];
+            } else {
+              shouldDisplayFallbackMessage = true;
+            }
+          } else {
+            const unknownErrorKeys = errorDetailKeys.filter(errorKey => {
+              return !(knownErrorKeys || []).includes(errorKey);
+            });
+
+            if (unknownErrorKeys.length) {
+              message = `${unknownErrorKeys[0]}: ${
+                errorDetail[unknownErrorKeys[0]]
```

Comment:
That same recursive function should be used here that extract a message from `unknownErrorKeys[0]` because it may not be a simple string. The function would take `unknownErrorKeys[0]` as argument and check and extract the first logical error message.